app/upload/page.tsx

```diff
@@ -5,91 +5,362 @@ import { Input } from '@/components/ui/input';
 import { Label } from '@/components/ui/label';
 import { useState } from 'react';
 import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { useAccount } from 'wagmi';
+import { useRouter } from 'next/navigation';
+import React, { useEffect } from 'react';
+
+
+const validateImageFile = (file: File | null, inputElement: HTMLInputElement): boolean => {
+  if (!file) return true;
+  
+  if (!file.type.startsWith('image/')) {
+    alert('Please select an image file');
+    inputElement.value = '';
+    return false;
+  }
+  
+  if (file.size > 5 * 1024 * 1024) {
+    alert('Image size must be less than 5MB');
+    inputElement.value = '';
+    return false;
+  }
+  
+  return true;
+};
 
 export default function Upload() {
-  const [selectedFile, setSelectedFile] = useState(null);
-  const [selectedAfterFile, setSelectedAfterFile] = useState(null);
-  const [beforeAfterImageSrc, setBeforeAfterImageSrc] = useState<string>();
-  const [beforeAfterImageDescription, setBeforeAfterImageDescription] =
-    useState<string>();
-
-  const onBeforeAfterImageSubmit = async (
-    event: React.FormEvent<HTMLFormElement>
-  ) => {
-    event.preventDefault();
+  const { address } = useAccount();
+  const router = useRouter();
+  const [beforeFile, setBeforeFile] = useState(null);
+  const [afterFile, setAfterFile] = useState(null);
+  const [description, setDescription] = useState('');
+  const [beforeImageSrc, setBeforeImageSrc] = useState<string>();
+  const [afterImageSrc, setAfterImageSrc] = useState<string>();
+  const [analysis, setAnalysis] = useState<string>();
+  const [isUploading, setIsUploading] = useState(false);
+  const [isAnalyzing, setIsAnalyzing] = useState(false);
+  const [currentImageId, setCurrentImageId] = useState<number>();
+
+  // Add confirmation before leaving page with unsaved changes
+  useEffect(() => {
+    const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+      if (beforeFile || afterFile) {
+        const message = 'You have unsaved changes. Are you sure you want to leave?';
+        e.preventDefault();
+        e.returnValue = message;
+        return message;
+      }
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+    return () => window.removeEventListener('beforeunload', handleBeforeUnload);
+  }, [beforeFile, afterFile]);
```

Comment:
@coderabbitai Im a bit confused here. When is handleBeforeUnload called and whats the problem with it?

Reply:
@coderabbitai Ok, so instead of removing the event listener, couldnt I just set beforeFile and afterFile to null once its uploaded inside the handleSubmit function?

Reply:
@coderabbitai so how would i set it to null? Just do beforeFile = null; and afterFile = null; or do I have to use a usestate or something like that